File: common/src/main/java/org/hiero/mirror/common/domain/transaction/AccessListEntry.java
// SPDX-License-Identifier: Apache-2.0

package org.hiero.mirror.common.domain.transaction;

import java.util.List;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@NoArgsConstructor
@SuperBuilder(toBuilder = true)
public class AccessListEntry {
    private String address;
    private List<String> storageKeys;
}

File: rest/model/accessListEntry.js
// SPDX-License-Identifier: Apache-2.0

import * as utils from '../utils.js';

class AccessListEntry {
  constructor(accessListEntry) {
    this.address = utils.toHexString(utils.stripHexPrefix(accessListEntry.address), true, 40);
    const storageKeys = accessListEntry.storage_keys ?? accessListEntry.storageKeys ?? [];
    this.storage_keys = storageKeys.map((key) => utils.toHexString(utils.stripHexPrefix(key), true, 64));
  }

  static ADDRESS = `address`;
  static STORAGE_KEYS = `storage_keys`;
}

export default AccessListEntry;
